nntrainer/compiler/tflite_interpreter.cpp

```diff
@@ -158,36 +180,62 @@ class TfOpIdxMap {
     buffer_map.addDataWhenNotFound(
       empty_buffer, {0, empty_buffer}); /// this represents empty buffer
 
-    auto &variable_map = getIndexMap<const float *, const Tensor *>();
-    auto update_variables = [&variable_map,
-                             &buffer_map](const TfOpNode::Variables &variables,
-                                          bool update_buffer) {
-      for (auto &variable : variables) {
-        const float *buf = variable->getData();
-        variable_map.addDataWhenNotFound(buf, variable);
-        if (update_buffer) {
+    auto update_buffer_map =
+      [&buffer_map](const TfOpNode::Variables &variables) {
+        for (auto &variable : variables) {
+          const float *buf = variable->getData();
+          if (variable->getName().find("nntrainer_internal_perm") !=
+              std::string::npos) {
+            buffer_map.addDataWhenNotFound(buf, {variable->bytes(), buf});
+          }
           buffer_map.addDataWhenNotFound(buf, {variable->bytes(), buf});
         }
-      }
-    };
+      };
+
+    auto register_tensors =
+      [&tensors = this->tensors](const TfOpNode::Variables &variables) {
+        for (auto &variable : variables) {
+          auto tensor_it = std::find(tensors.begin(), tensors.end(), variable);
+          if (tensor_it == tensors.end()) {
+            tensors.push_back(variable);
+          }
+        }
+      };
 
     for (auto &op_node : nodes) {
+      if (op_node->isVirtualNode())
+        continue;
       update_opcode(op_node->getOpType());
-      update_variables(op_node->getInputs(), false);
-      update_variables(op_node->getWeights(), true);
-      update_variables(op_node->getOutputs(), false);
+      /**
+       * Q) Why only add graph input?
+       *
+       * A) the tflite needs only one tensor between nodes. Therefore,
+       *basically, no inputs are considered except graph input that doesn't have
+       *FROM node.
+       **/
+      if (op_node->isInputNode()) {
+        register_tensors(op_node->getInputs());
+        update_buffer_map(op_node->getInputs());
+      }
+      register_tensors(op_node->getWeights());
+      register_tensors(op_node->getOutputs());
+
+      update_buffer_map(op_node->getWeights());
```

Comment:
Do we only care about the buffer of weights? I mean it does not need to concern with the buffer of input and output tensor?

Reply:
@jijoongmoon Good point. After a few surveys, I realized we do have to care about the buffer of others. I mean, each output tensors have to got its own buffer.

So, I'm going to add extra `register_tensors` call for `op_node->getOutputs()`.